Support updating mobile user roles via the API

gherceg opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
Adds support to update user roles for mobile users via the API.
## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi-dev.atlassian.net/browse/SAAS-12428

Prior to updating the API to reject requests to update the user role (due to being an unknown field), this was not functioning properly for mobile users. This is no longer relevant for web users since the API to create/edit web users has been removed.

This just adds a method to update user roles for the key/field `role`. I did notice a method `_set_role_for_bundle` that I may have clobbered in recent refactors. I chose to raise an exception if the ambiguous case were hit (multiple roles with the same name within the domain), rather than just update the first one, but am open to feedback there.

### Design Question

~This is incomplete, as I should still add tests in `TestCommCareUserResourceUpdate` verifying the newly added error messages if I were to continue.~

The reason I'm waiting is because I don't like the direction this is heading. Right now, the `_update` method in the top level user api file (v0_5.py) calls an internal update method, handles specific exceptions raised, and translates those exceptions into user facing error messages. While this separation seemed nice at first, I think it would be better to move the error message generation down into the `user_updates.py` file to allow for more specific error messages, without overcrowding the caller that needs to know how to handle all of these exceptions. This would reduce the number of custom exceptions needed/used as well.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
Added automated tests for this new method.
### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
add method to update mobile user roles via api

snopoke reviewed (commented):
Changes look good so far. Regarding the design question, I think your suggestion seems fine, particularly if there are multiple places that need to be able to translate the exceptions into user facing error messages.

gherceg committed:
remove unnecessary fetch

gherceg commented:
I rebased because an adjacent change to the api was made, and I'm not too far into this work yet. I didn't edit any commits. You'll see I added 9dd8bd397ad29bc8e730594d660ddf81f1509263 to address the minor test feedback, and am working on a simpler design for this today to address my issues.

gherceg committed:
raise more general exception with specific message for error

gherceg committed:
fix failing test

gherceg committed:
run make translations

gherceg commented on corehq/apps/api/tests/test_user_updates.py:
```diff
@@ -130,6 +133,29 @@ def test_update_unknown_field_raises_exception(self):
+            self.domain, 'edit-data', permissions=Permissions(edit_data=True)
+        )
+        update(self.user, 'role', 'edit-data')
+        roles = UserRole.objects.by_domain_and_name(self.domain, 'edit-data')
```
see 9dd8bd397ad29bc8e730594d660ddf81f1509263

millerdev commented on corehq/apps/api/tests/test_user_updates.py:
```diff
@@ -45,23 +42,25 @@ def test_update_password_without_strong_passwords_succeeds(self):
             update(self.user, 'password', 'abc123')
-        except ValidationError:
+        except UpdateUserException:
             self.fail('Unexpected ValidationError raised.')
```
nit
```suggestion
            self.fail('Unexpected UpdateUserException raised.')
```
An alternate suggestion would be to remove the try/except and possibly add a comment:
```py
update(self.user, 'password', 'abc123')  # should not raise
```

gherceg committed:
simplify tests that assert exception is not raised
